CONTRIBUTING.md

```diff
@@ -1,81 +1,120 @@
 
-##### Running Tests
+The tests for the fetch service require a local httpbin server. We recommend running this with [Docker](https://www.docker.com/):
 
 ```bash
-./ci/run_tests.sh
+docker run -p 8000:80 kennethreitz/httpbin
 ```
-or
+
+Before running the tests you need to set the `HTTPBIN_URL` environment variable to the url of your httpbin instance.
+If you used the previous command to start the server the value should be "http://localhost:8000" (It's important that you don't add a trailing slash).
+
+```shell
+# Unix-like
+export HTTPBIN_URL="http://localhost:8000"
+
+# Windows
+set HTTPBIN_URL=http://localhost:8000
+```
+
+You also need to activate the `httpbin_test` feature in order for the tests to run:
 
 ```bash
-cargo test --target wasm32-unknown-unknown --features wasm_test
+wasm-pack test --firefox --headless -- --features wasm_test,httpbin_test
 ```
 
-#### Benchmarks
+If you're using the `/ci/run_tests.sh` script you only need to set the environment variable. The script will automatically add the feature.
+
+### Macro Tests
+
+When adding or updating tests, please make sure you have updated the appropriate `stderr` file, which you can find [here](https://github.com/yewstack/yew/tree/master/yew-macro/tests/macro) for the `html!` macro. These files ensure that macro compilation errors are correct and easy to understand.
+
+To update or generate a new `stderr` file you can run `TRYBUILD=overwrite cargo test --test macro_test` or `TRYBUILD=overwrite cargo test --test derive_props_test` from the `yew-macro` directory.
+
+## Benchmarks
 
 If you wish to improve the performance of Yew, we ask you to prove the improvements of your changes through benchmarking.
 
-1. Fork and clone https://github.com/yewstack/js-framework-benchmark
+1. Fork and clone <https://github.com/yewstack/js-framework-benchmark>
 2. Update `frameworks/yew/Cargo.toml` with your fork of Yew and the branch for your changes
 3. Open a new PR with your `Cargo.toml` changes
 
 Feel free to add new benchmark tests if the current benchmark coverage is insufficient!
 
-### Writing APIs
+## Writing APIs
+
 When building new APIs, think about what it would be like to use them. Would this API cause confusing and hard to pin error mesages? Would this API integrate well with other APIs? Is it intuitive to use this API?
 
-There are many resources which provide good guidance on how to write APIs, a few of which are given below. These are only *guidlines* and while they are useful and should be followed where possible, in some cases it may not be possible to do so.
-* [The Rust API Guidelines](https://rust-lang.github.io/api-guidelines/)
-* [Elegant Library APIs in Rust](https://deterministic.space/elegant-apis-in-rust.html)
+There are many resources which provide good guidance on how to write APIs, a few of which are given below. These are only _guidlines_ and while they are useful and should be followed where possible, in some cases it may not be possible to do so.
```

Comment:
This was actually a typo of mine, but now seems like a good time to fix it.
```suggestion
There are many resources which provide good guidance on how to write APIs, a few of which are given below. These are only _guidelines_ and while they are useful and should be followed where possible, in some cases it may not be possible to do so.
```